Sources/MapboxMaps/Location/InterpolatedLocationProducer.swift

```diff
@@ -0,0 +1,89 @@
+import CoreLocation
+
+internal protocol InterpolatedLocationProducerProtocol: AnyObject {
+    var location: InterpolatedLocation? { get }
+    func observe(with handler: @escaping (InterpolatedLocation) -> Bool) -> Cancelable
+}
+
+internal final class InterpolatedLocationProducer: NSObject, InterpolatedLocationProducerProtocol {
+    private var startDate: Date?
+    private var endDate: Date?
+    private var startLocation: InterpolatedLocation?
+    private var endLocation: InterpolatedLocation?
+
+    private let observableInterpolatedLocation: ObservableInterpolatedLocationProtocol
+    private let locationInterpolator: LocationInterpolatorProtocol
+    private let dateProvider: DateProvider
+
+    internal var location: InterpolatedLocation? {
+        observableInterpolatedLocation.value
+    }
+
+    internal init(observableInterpolatedLocation: ObservableInterpolatedLocationProtocol,
+                  locationInterpolator: LocationInterpolatorProtocol,
+                  dateProvider: DateProvider,
+                  locationProducer: LocationProducerProtocol,
+                  displayLinkCoordinator: DisplayLinkCoordinator) {
+        self.observableInterpolatedLocation = observableInterpolatedLocation
+        self.locationInterpolator = locationInterpolator
+        self.dateProvider = dateProvider
+        super.init()
+        locationProducer.add(self)
```

Comment:
Adding self as consumer for location producer instantly starts location updates.
That's said MapView will always start location updates using CLLocationManager event if user's location is not shown on map

Reply:
https://github.com/mapbox/mapbox-maps-ios/issues/1251

Reply:
@rkapshuk-ring the fix for this issue was released today in v10.4.3.